### PR TITLE
Fix physics (trimesh) positioning

### LINKED ISSUE #62: TriMesh physics offset issues
I've just started playing with enable3d, and I really like it. When going through the Fantasy Book example, I noticed the comments and code regarding "unexpected offset" in the physics concave meshes. I did a little research and I found that the offset is exactly the mesh's parent object location ;)

The problem as far as I can see, is the code in [_iterateGeometries](https://github.com/enable3d/enable3d/blob/41b742749073fe819b737ed37eab5985d95241db/packages/ammoPhysics/src/three-to-ammo.js#L681) which builds the transform assuming the "root" has the identity matrix as transform, since the body will get the local position/rotation of the "root".

However, this is not correct. If any of its parents has any transform (e.g. position/rotation set) these will not be taken into account in calculating the world transform for the physics mesh, resulting in offset/rotated physics.

My matrix math sucks, so I haven't come up with a fix for it yet, but I thought I'd share, for any other people running into weird positioning issues with this.

### PR DESCRIPTION
**Pull request by Ithamar**
This PR fixes #62, see the issue for a detailed description of the problem.

### CODE REVIEW

**Comment by Ithamar** [comment]
Please note that I have not finished full testing yet, there are likely still some edge cases that do not work properly. However, the main case as described in #62 works fine, even with a dynamic body.

**Comment by Ithamar** [comment]
Okay, I think this should be it, works for all cases I have been able to come up with (compound/single shape, scaling, and both transforms on mesh level as well as parent, and body types dynamic/static)..
Feel free to comment away ;)

**Comment by yandeu** [comment]
Nice! Thanks a lot!

I will build the bundle and check if it works with the examples on the website and if it fixes issue #39, #47 and #48.

**Comment by Ithamar** [comment]
ah yes, that is a good point, I should have thought of testing that way. Please let me know if you run into any unexpected breakage in those, I'm more then happy to debug.

**Comment by yandeu** [comment]
Looks like all examples still work (except for a small issue with three-to-ammo but it is not related to this fix). Even the "Medieval Fantasy Book" and issue 47 work. (Could not test issue 39 and 48, since I do not have the models).

Thanks very much for your fix!

File: packages/ammoPhysics/src/physics.ts [hunk 1]
```diff
@@ -104,6 +107,9 @@ class AmmoPhysics extends EventEmitter {
     this.tmpEuler = new Euler()
     this.tmpQuaternion = new Quaternion()
     this.tmpVector3 = new Vector3()
+    this.tmpVector3a = new Vector3()
+    this.tmpMatrix4 = new Matrix4();
+    this.tmpMatrix4a = new Matrix4();
```
Comment (yandeu) [inline packages/ammoPhysics/src/physics.ts:112]:
```suggestion
    this.tmpMatrix4 = new Matrix4()
    this.tmpMatrix4a = new Matrix4()
```

File: packages/ammoPhysics/src/physics.ts [hunk 2]
```diff
@@ -297,15 +303,25 @@ class AmmoPhysics extends EventEmitter {
...
+          this.tmpVector3a.set(scale.x(), scale.y(), scale.z())
+          this.tmpVector3.set(p.x() + o.x, p.y() + o.y, p.z() + o.z)
+          this.tmpQuaternion.set(q.x(), q.y(), q.z(), q.w())
+          this.tmpMatrix4.compose(this.tmpVector3, this.tmpQuaternion, this.tmpVector3a)
+          if (objThree.parent) {
+            this.tmpMatrix4a.getInverse(objThree.parent.matrixWorld)
+          } else {
+            this.tmpMatrix4a.identity()
+          }
+          this.tmpMatrix4a.multiply(this.tmpMatrix4);
```
Comment (yandeu) [inline packages/ammoPhysics/src/physics.ts:323]:
```suggestion
          this.tmpMatrix4a.multiply(this.tmpMatrix4)
```

**Pull request merged**